### ISSUE TITLE
the tflite file generated by export_tflite.py did not work correctly

### ISSUE THREAD

**Issue by spark123456789** [sercant/mobile-segmentation#10]
Dear Sir,
         Thank you for sharing your code, It do help me a lot.
         The training and testing part of the model work very well.
         But when I start to generate a pb and tflite file, problem comes.
         Right now, I use tensorflow with a version of 2.0 beta. Just modify the directory of logs in order to execute the conversion script. As I look deeper into the code, the input node is "input_0" , the output node is "Cast" , which convert the data from int64 to int32. I also notices that, the outputs_to_scales_to_logits  is generated by model.predict_labels, which include a operator "ArgMax", it also supported by the latest version of tflite. But the conversion progress did not report any errors, so I test the generated tflie file by the scripy below:

import numpy as np
import tensorflow as tf

# Load TFLite model and allocate tensors.
interpreter = tf.lite.Interpreter(model_path="segmentation.tflite")
interpreter.allocate_tensors()

# Get input and output tensors.
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

# Test model on random input data.
input_shape = input_details[0]['shape']
print(input_shape)
input_data = np.array(np.random.random_sample(input_shape), dtype=np.float32)
print(input_data)
interpreter.set_tensor(input_details[0]['index'], input_data)

interpreter.invoke()
output_data = interpreter.get_tensor(output_details[0]['index'])
print(np.unique(output_data))

I just used a random picture to do the test, willing to see different labels [total 14 classes in my case] sorted by np.unique function. but unfortunately,  just one number[4] or two numbers[3,4] is shown. Can you give me some ideas how to solve this problem? Thank you very much!

I also test the tflite file with a test picture, it also did not report expected results.

**Comment by sercant** [comment]
Hello @spark123456789,

Happy that the repository has been a good source for you. I haven't tested with Tensorflow 2.0 yet but there are some things that you can try. Did you try visualizing the result? Also, can you try running the inference with the frozen `.pb` file and see if it is correct?

Other things to keep in mind is if you didn't use `DPC` during the training, `USE_DPC` variable in the `export_tflite.py` should be set to `False`. Is your background class marked as `0` in your dataset? Because with the random data, I would expect that you mostly get background class in the result.

Let me know if any of these give you some progress or not.

**Comment by spark123456789** [comment]
Dear Sir,
Below is the script I used to test the frozen you provided on the website.

import numpy as np
import tensorflow as tf
from PIL import Image

if __name__ == '__main__':

    image     = Image.open("default.jpg")

    image = image.resize((769, 769), Image.ANTIALIAS)

    image = image.convert("RGB")
	
    image = np.asarray(image)
    prepimg = (image -127.5)/ 127.5


    with tf.gfile.FastGFile("frozen_graph.pb", "rb") as f:
        graphdef = tf.GraphDef()
        graphdef.ParseFromString(f.read())
        _ = tf.import_graph_def(graphdef, name="")

    with tf.Session() as sess:

        tf.initialize_all_variables().run()

        input_x = sess.graph.get_tensor_by_name("input_0:0")        
        output = sess.graph.get_tensor_by_name("Cast:0")

        predictions = sess.run(output, {input_x:[prepimg]})
        print(np.unique(predictions))

the default.jpg is an image from CityScape .
The output is only [0 13], which did not related to the input image totally.
Is there any mistake in my test script? or What I can further test in my condition? 
Thank you very much!

**Comment by spark123456789** [comment]
Here is the default image link:
https://www.google.com/search?q=cityscape+dataset&safe=active&source=lnms&tbm=isch&sa=X&ved=0ahUKEwitm5ygmr7jAhWJUN4KHQocAd4Q_AUIESgB&biw=1237&bih=623&dpr=1.5#imgrc=wha2pUXwEAggSM:

**Comment by sercant** [comment]
Hello @spark123456789,

The preprocessing is included in the graph. Can you try after removing the `prepimg = (image -127.5)/ 127.5` line? Here is a script that I have just tried.

**Code**
```python
import tensorflow as tf
import numpy as np
from PIL import Image

im = Image.open('default.png')

arr = np.array(im.convert('RGB'))

with tf.gfile.GFile("frozen_graph.pb", "rb") as f:
    graphdef = tf.GraphDef()
    graphdef.ParseFromString(f.read())
    tf.import_graph_def(graphdef, name="")

with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    sess.run(tf.local_variables_initializer())

    input_x = sess.graph.get_tensor_by_name("input_0:0")
    output = sess.graph.get_tensor_by_name("Cast:0")

    predictions = sess.run(output, {input_x: [arr]})
    print(np.unique(predictions))
```

**Input Image**

![default](https://user-images.githubusercontent.com/6319501/61449820-6c451f00-a95e-11e9-8936-2e6ca2e26f4e.png)

**Output**

`[ 0  1  2  3  4  7  8  9 10 13 14 15 18]`

**Comment by sercant** [comment]
I'm closing the issue since there hasn't been any activity for a long time.
